services/horizon/internal/reap/system.go

```
@@ -70,19 +70,23 @@ func (r *System) runOnce(ctx context.Context) {
 	}
 }
 
-// Work backwards in 100k ledger blocks to prevent using all the CPU.
+// Work backwards in 50k (by default, otherwise configurable via the CLI) ledger
+// blocks to prevent using all the CPU.
 //
-// This runs every hour, so we need to make sure it doesn't
-// run for longer than an hour.
+// This runs every hour, so we need to make sure it doesn't run for longer than
+// an hour.
 //
-// Current ledger at 2021-08-12 is 36,827,497, so 100k means 368 batches. At 1
-// batch/second, that seems like a reasonable balance between running well
-// under an hour, and slowing it down enough to leave some CPU for other
-// processes.
-var batchSize = int32(100_000)
+// Current ledger at 2024-04-04s is 51,092,283, so 50k means 1021 batches. At 1
+// batch/second, that seems like a reasonable balance between running under an
+// hour, and slowing it down enough to leave some CPU for other processes.
 var sleep = 1 * time.Second
 
 func (r *System) clearBefore(ctx context.Context, startSeq, endSeq int32) error {
+	batchSize := int32(r.RetentionBatch)
+	if batchSize <= 0 {
+		batchSize = int32(50_000)
```

minor nit for consideration, pull this up to a package const, `reap_batch_default_size = 50_000`

---

since you have already configured the default in flags.go I would just return an error in case `batchSize <= 0` because that should be unexpected. you can also add validation in flags.go to ensure the batch size must be at least 1

---

yeah fair enough, done :+1: